CMakeLists.txt

```
@@ -0,0 +1,67 @@
+cmake_minimum_required(VERSION 3.22)
+
+project(ClpIrV1Decoder)
+
+# Download and extract Boost
+set(BOOST_VERSION "1.85.0")
+string(REPLACE "." "_" BOOST_ARCHIVE_VERSION_PART ${BOOST_VERSION})
+
+include(ExternalProject)
+ExternalProject_Add(
+        Boost
+        URL "https://archives.boost.io/release/${BOOST_VERSION}/source/boost_${BOOST_ARCHIVE_VERSION_PART}.tar.gz"
+        PREFIX "${CMAKE_BINARY_DIR}/boost"
+        CONFIGURE_COMMAND ""
+        BUILD_COMMAND ""
+        INSTALL_COMMAND ""
+        LOG_DOWNLOAD ON
+)
+ExternalProject_Get_Property(Boost SOURCE_DIR)
+set(BOOST_EXTRACT_DIR "${SOURCE_DIR}")
+
```

Let's separate the target definition and source file additions from a single `add_executable` from below.

```suggestion

set(BIN_NAME "ClpIrV1Decoder" CACHE STRING "Binary name for the generated .js file and .wasm file of CLP IR V1 Decoder.")
add_executable(${BIN_NAME})
```